nodejs/aws-infra/experimental/ecs/taskDefinition.ts

```diff
@@ -195,38 +156,32 @@ export abstract class TaskDefinition extends pulumi.ComponentResource {
     }
 }
 
-function createRunFunction(
-        isFargate: boolean,
-        taskDefArn: pulumi.Output<string>,
-        containerToEnvironment: pulumi.Output<Record<string, aws.ecs.KeyValuePair[]>>) {
+export type RunTaskRequest = utils.Overwrite<awssdk.ECS.RunTaskRequest, {
+    /**
+     * The Cluster to run the Task in.
+     */
+    cluster: ecs.Cluster;
+    taskDefinition: never;
+    launchType: never;
+}>;
+
+function createRunFunction(isFargate: boolean, taskDefArn: pulumi.Output<string>) {
+    return function runTask(
+                params: RunTaskRequest,
+                callback?: (err: awssdk.AWSError, data: awssdk.ECS.Types.RunTaskResponse) => void,
+            ): awssdk.Request<awssdk.ECS.Types.RunTaskResponse, awssdk.AWSError> {
 
-    return async function runTask(options: TaskRunOptions) {
         const ecs = new aws.sdk.ECS();
 
-        const cluster = options.cluster;
+        const cluster = params.cluster;
         const clusterArn = cluster.instance.id.get();
         const securityGroupIds = cluster.securityGroups.map(g => g.instance.id.get());
         const subnetIds = cluster.vpc.publicSubnetIds.map(i => i.get());
-
-        const innerContainers = containerToEnvironment.get();
-        const containerName = options.containerName || Object.keys(innerContainers)[0];
-        if (!containerName) {
-            throw new Error("No valid container name found to run task for.");
-        }
-
-        // Extract the environment values from the options
-        const env1 = innerContainers[containerName] || [];
-        const env2 = options.environment || [];
-
-        const env = [...env1, ...env2];
-
         const assignPublicIp = isFargate; // && !usePrivateSubnets;
```

Comment:
This change to remove `usPrivateSubnets` was made in the recent change from Network to Vpc.  I'm not sure this is correct now - Tasks in a public+private network should be running in the private subnet and should not be assigned public IPs.

Reply:
yup.  this def isn't good.  i will address as part of the work to allow vpcplacement's to be provided.